mhkit/dolfyn/adp/clean.py

```diff
@@ -259,16 +267,26 @@ def water_depth_from_pressure(ds, salinity=35) -> None:
             "The variable 'depth' already exists. "
             "Please manually remove 'depth' if it needs to be recalculated."
         )
-    if "pressure" not in ds.data_vars:
+    pressure = [v for v in ds.data_vars if "pressure" in v]
+    if not pressure:
         raise NameError("The variable 'pressure' does not exist.")
-    elif not ds["pressure"].sum():
-        raise ValueError("Pressure data not recorded.")
-    if "temp" not in ds.data_vars:
+    else:
+        for p in pressure:
+            if not ds[p].sum():
+                pressure.remove(p)
+        if not pressure:
+            raise ValueError("Pressure data not recorded.")
+    temp = [
+        v
+        for v in ds.data_vars
+        if (("temp" in v) and ("clock" not in v) and ("press" not in v))
+    ]
+    if not temp:
         raise NameError("The variable 'temp' does not exist.")
 
     # Density calcation
-    P = ds["pressure"].values
-    T = ds["temp"].values  # temperature, degC
+    P = ds[pressure[0]].values  # pressure, dbar
```

Comment:
do all instruments use dbar (over Pa)? or is it well described in the examples? Should this be added to the docstring?

Reply:
Yes, they all use dbar because it translates nearly 1:1 as meters-beneath-the-surface (assuming the pressure sensor was zeroed before deploying, of course)